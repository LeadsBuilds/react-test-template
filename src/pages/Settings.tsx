import AttendeeList from "../components/AttendeeList";
import Footer from "../components/Footer";
import Form from "../components/Form";

const Settings = () => {
    return (
        <>
           <Form/>
           <AttendeeList/>
           <Footer/> 
        </>
    );
}

export default Settings;