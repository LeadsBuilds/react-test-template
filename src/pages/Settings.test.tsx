import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import Settings from "./Settings";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavigate
    }
});

describe('The settings page', () => {
    test('Must be rendered correctly', () => {
        const {container} = render(
            <RecoilRoot>
                <Settings/>
            </RecoilRoot>
        );

        expect(container).toMatchSnapshot();
    });
});