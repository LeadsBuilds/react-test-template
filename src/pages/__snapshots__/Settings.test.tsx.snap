// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The settings page Must be rendered correctly 1`] = `
<div>
  <form>
    <input
      placeholder="Type here the name of the attendees"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Add
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Start game
    </button>
  </footer>
</div>
`;
